### PR TITLE
Update signAndSubmitBatch

### PR DESCRIPTION
**Pull request by Lanikean**

### CODE REVIEW

File: provider/grpc.go [hunk 1]
```diff
@@ -283,6 +283,31 @@ func (g *GRPCClient) signAndSubmitBatch(ctx context.Context, transactions []*pb.
...
+
+	if len(transactions) == 1 {
+		txBase64, err := transaction.SignTxWithPrivateKey(transactions[0].Content, *g.privateKey)
+		if err != nil {
+			return nil, err
+		}
+
+		res, err := g.PostSubmit(ctx, &pb.TransactionMessage{
+			Content:   txBase64,
+			IsCleanup: false,
```
Comment (tinystarinagalaxy) [inline provider/grpc.go:295]: plz make sure the `isCleanup`, `frontRunningProtection` and `useStakedRPCs` come from the request

Comment (tinystarinagalaxy) [reply provider/grpc.go:295]: you need to do it for http and ws as well

File: provider/ws.go [hunk 2]
```diff
@@ -541,6 +542,28 @@ func (w *WSClient) SignAndSubmitBatch(ctx context.Context, transactions []*pb.Tr
...
 		return nil, ErrPrivateKeyNotFound
 	}
+
+	if len(transactions) == 1 {
+		txBase64, err := transaction.SignTxWithPrivateKey(transactions[0].Content, *w.privateKey)
+		if err != nil {
+			return nil, err
+		}
+
+		res, err := w.PostSubmit(ctx, txBase64, *opts.SkipPreFlight, false, false)
```
Comment (tinystarinagalaxy) [inline provider/ws.go:552]: this `false,false` need to come from the request

File: provider/http.go [hunk 3]
```diff
@@ -461,6 +461,28 @@ func (h *HTTPClient) SignAndSubmitBatch(ctx context.Context, transactions []*pb.
...
 		return nil, ErrPrivateKeyNotFound
 	}
+
+	if len(transactions) == 1 {
+		txBase64, err := transaction.SignTxWithPrivateKey(transactions[0].Content, *h.privateKey)
+		if err != nil {
+			return nil, err
+		}
+
+		res, err := h.PostSubmit(ctx, txBase64, *opts.SkipPreFlight, false, false)
```
Comment (tinystarinagalaxy) [inline provider/http.go:471]: this `false,false` need to come from the request

File: provider/grpc.go (hunk 1 above)
Comment (DimaCantemir) [reply provider/grpc.go:295]: In this case IsCleanup will be FALSE always...so it's fine

**Review by DimaCantemir — APPROVED** [review]
LGTM

**Pull request merged**